mesonbuild/backend/xcodebackend.py

```diff
@@ -1802,9 +1802,7 @@ def generate_single_build_target(self, objects_dict: PbxDict, target_name: str,
                 # Xcode uses GCC_PREFIX_HEADER which only allows one file per target/executable. Precompiling various header files and
                 # applying a particular pch to each source file will require custom scripts (as a build phase) and build flags per each
                 # file. Since Xcode itself already discourages precompiled headers in favor of modules we don't try much harder here.
-                pchs = target.get_pch('c') + target.get_pch('cpp') + target.get_pch('objc') + target.get_pch('objcpp')
-                # Make sure to use headers (other backends require implementation files like *.c *.cpp, etc; these should not be used here)
-                pchs = [pch for pch in pchs if pch.endswith('.h') or pch.endswith('.hh') or pch.endswith('hpp')]
+                pchs = [t[0] for t in target.pch['c'] + target.pch['cpp'] if t is not None]
```

Comment:
Is `+` correct here as opposed to `[target.pch['c'], target.pch['cpp']]`? It would concatenate the tuples, I think.

Reply:
yeah, you're right. I'll fix that.